File: src/numeric/primes500/primes.ts

function* prime_sieve() : Generator<number, void, any> {
    yield 2;
    let cache = [2];
    var candidate = 3;
    while (true) {
        if (cache.every((prime) => candidate % prime != 0)) {
            yield candidate;
            cache.push(candidate);
        }

        candidate += 2;
    }
}

function get_primes(count: number): Array<number> {
    let primes = [];
    for (const prime of prime_sieve()) {
        primes.push(prime);
        if (primes.length >= count) {
            break;
        }
    }
    return primes;
}

function print_primes(primes: Array<number>) {
    console.log("First Five Hundred Primes");
    var pad = (n: number) => n.toString().padStart(4, '0');
    for (var n = 0; n < 50; ++n) {
        console.log(`     ${pad(primes[n])} ${pad(primes[n + 50])}`
            + ` ${pad(primes[n + 100])} ${pad(primes[n + 150])}`
            + ` ${pad(primes[n + 200])} ${pad(primes[n + 250])}`
            + ` ${pad(primes[n + 300])} ${pad(primes[n + 350])}`
            + ` ${pad(primes[n + 400])} ${pad(primes[n + 450])}`);
    }
}

print_primes(get_primes(500));

File: src/numeric/primes500/primes.jl

function get_primes()
    primes = [2]
    n = 3

    for i = 1:500
        push!(primes, n)

        is_prime = false
        while !is_prime
            n += 2

            q = 9999
            r = 1
            t = 0
            k = 2
            while (r != 0) && (q > t)
                t = primes[k]
                q = n / t
                r = n % t
                k += 1
            end

            is_prime = (r != 0) && (q <= t)
        end
    end
    return primes
end

function print_primes(primes)
    println("First Five Hundred Primes")
    for i = 1:50
        println("     ", string(primes[i], pad=4), " ",
            string(primes[50 + i], pad=4), " ",
            string(primes[100 + i], pad=4), " ",
            string(primes[150 + i], pad=4), " ",
            primes[200 + i], " ", primes[250 + i], " ",
            primes[300 + i], " ", primes[350 + i], " ",
            primes[400 + i], " ", primes[450 + i])
    end
end

print_primes(get_primes())
